Mathlib/CategoryTheory/Limits/Shapes/Kernels.lean

```diff
@@ -214,6 +214,35 @@ lemma KernelFork.IsLimit.isIso_ι {X Y : C} {f : X ⟶ Y} (c : KernelFork f)
 
 end
 
+namespace KernelFork
+
+variable {f} {X' Y' : C} {f' : X' ⟶ Y'}
+
+/-- The morphism between points of kernel forks induced by a morphism
+in the category of arrows. -/
+def mapOfIsLimit (kf : KernelFork f) {kf' : KernelFork f'} (hf' : IsLimit kf')
+    (φ : Arrow.mk f ⟶ Arrow.mk f') : kf.pt ⟶ kf'.pt :=
+  hf'.lift (KernelFork.ofι (kf.ι ≫ φ.left) (by simp))
+
+@[reassoc (attr := simp)]
+lemma mapOfIsLimit_ι (kf : KernelFork f) {kf' : KernelFork f'} (hf' : IsLimit kf')
+    (φ : Arrow.mk f ⟶ Arrow.mk f') :
+    kf.mapOfIsLimit hf' φ ≫ kf'.ι = kf.ι ≫ φ.left :=
+  hf'.fac _ _
+
+/-- The isomorphism between points of limit kernel forks induced by an isomorphism
+in the category of arrows. -/
+@[simps]
+def mapIsoOfIsLimit {kf : KernelFork f} {kf' : KernelFork f'}
```

Comment:
Can we deduce this from an isomorphism of `KernelFork`s instead? If not, can we have (or have a TODO for) the isomorphism, with a statement that getting the isomorphism of their points is the same as this one.

Reply:
These two kernel forks are not in the same category (one is in `KernelFork f`, and the other in `KernelFork f'`), so that we cannot directly use `IsLimit.conePointUniqueUpToIso`. The definitions I have done are very parallel to that of `kernel.map` and `kernel.mapIso`: I only generalize to arbitrary limit kernel forks because it will be necessary for some computations in group cohomology. It should be possible to refactor the functoriality of kernels (either the chosen `kernel`, or the points of limit kernels forks) by constructing a functor `Arrow C ⥤ (WalkingParallelPair ⥤ C)` and then studying the associated functoriality of cones, but I am not sure I would like to encourage anybody to pursue this direction.